internal/cosesign1/misc.go

```diff
@@ -0,0 +1,152 @@
+package cosesign1
+
+//	Little handy utilities that make logging and a command line tool easier.
+
+import (
+	"fmt"
+	"os"
+
+	"crypto/x509"
+	"encoding/base64"
+	"io"
+	"log"
+
+	"github.com/veraison/go-cose"
+)
+
+func ReadBlob(filename string) []byte {
+	content, err := os.ReadFile(filename)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return content
+}
+
+func ReadString(filename string) string {
+	content := ReadBlob(filename)
+	str := string(content)
+	return str
+}
+
+// Type to replace the rand.Reader with a source of fixed salt.
+// Can be provided to the cose.Sign1 method instead of rand.Reader such that
+// the signature is deterministic for testing and debugging purposes.
+type fixedReader struct {
+	valueToReturn byte
+}
+
+func (fr *fixedReader) Read(p []byte) (int, error) {
```

Comment:
hopefully this won't be called to read large chunks of data, otherwise it'll be painful to read 1 byte at a time.

Reply:
This is why we have optimisers :)

As per the comment, it is the code for test cases where a zero salt is to be introduced so generated stuff ends up the same every time. Typically, those are 32 or 64 bytes.